vispy/visuals/surface_plot.py

```diff
@@ -126,7 +126,7 @@ def _prepare_mesh_colors(self, colors):
             # convert (width, height, 4) to (num_verts, 4)
             vert_shape = self.__vertices.shape
             num_vertices = vert_shape[0] * vert_shape[1]
-            colors = colors.reshape(num_vertices, 3)
+            colors = colors.reshape(num_vertices, 4)
```

Comment:
There is another merge request with a fix for this that was never merged and has a similar fix:

https://github.com/vispy/vispy/pull/2620/files#diff-e962a5a20e008c521c34955c7cc159ff74e96a8c2edaf8a7bf22733c402c28f9

Mainly, instead of `4` it does `colors.shape[-1]` to handle RGB and RGBA colors.

```suggestion
            colors = colors.reshape(num_vertices, colors.shape[-1])
```